### ISSUE TITLE
Missing ref on ComposedChart in v2.13

### ISSUE THREAD

**Issue by luap2703** [recharts/recharts#5113]
## Do you want to request a feature or report a bug?
Bug/Request

## What is the current behavior?
When updating from 2.12 => 2.13 ref property on ComposedChart was removed. This is an issue as it is currently the only way to access the internal behavior of the chart (right?). What workarounds allow to access it again?

I saw https://github.com/recharts/recharts/issues/2665#issuecomment-2354007954 but it seems that those mentioned hooks aren't implemented yet, right? So now there's no way to access the internal state?

```tsx 
  <ComposedChart
                    ref={chartRef as any} // Ref is not available anymore
                    accessibilityLayer
                    // data={dataWithRange}
                    data={presentedDataset.data}
                    margin={{
                      top: 10,
                      right: 20,
                      left: 20,
                      bottom: 0,
                    }}
                    onMouseMove={onMouseMove}
                    onMouseUp={onMouseUp}
                    onMouseLeave={onMouseUp}
                  >
```

We need it to currently access:
```tsx
    const lowerBoundValue = (chartRef.current?.state.yAxisMap?.[yAxisId].domain as any[])[0];
          const upperBoundValue = (chartRef.current?.state.yAxisMap?.[yAxisId].domain as any[])[1];
          const diffValue = upperBoundValue - lowerBoundValue;

          const maxY = (chartRef.current?.state.prevHeight ?? diffValue) as number;
```

**Comment by stuartherring** [comment]
We had the same issue with `PieChart` - although it seems like it's only a typing issue, since the code itself still works as it did in 2.12.x, it's only Typescript that doesn't see the `ref` anymore.

**Comment by ckifer** [comment]
Sorry all, I'm on vacation without my computer so can't fix this for ~2 weeks. 

Not sure why or how these got touched. As mentioned it's just a TS error, so a TS-expect-error should work until it's fixed

**Comment by ckifer** [comment]
@PavelVanecek fyi not sure what happened here but if you feel like looking

**Comment by luap2703** [comment]
@ckifer @PavelVanecek  thank you for checking! Unfortunately, I have to confirm that it's not just a TS error, the ref got actually removed on ComposedChart (or at least some methods on the ref, I can double-check that if needed).

> Sorry all, I'm on vacation without my computer so can't fix this for ~2 weeks.
> 
> Not sure why or how these got touched. As mentioned it's just a TS error, so a TS-expect-error should work until it's fixed
> 
> And no, hooks are not ready yet

**Comment by ckifer** [comment]
Feel free to check, from my perspective and the diff nothing like that should have changed. I'll be back on the 28th

**Comment by ckifer** [comment]
Feel free to downgrade until I am able to investigate.

**Comment by sKopheK** [comment]
upgraded to overcome `defaultProps` issue but receiving this instead. we need to access `clipPathId`. 2.13 not only changed typings  (check types/charts directory e.g., https://www.npmjs.com/package/recharts/v/2.12.7?activeTab=code vs https://www.npmjs.com/package/recharts/v/2.13.0?activeTab=code) but at least `ref` prop is gone:  `Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?`
![{3125F5D9-4631-48E3-8E61-12FFF97CD480}](https://github.com/user-attachments/assets/49fe2ddc-15cd-4f88-8afb-a66b333cea41)

**Comment by ckifer** [comment]
I have a hunch as to why this happened, I'll look at this today at some point

**Comment by ckifer** [comment]
I fixed a typescript issue for React 19 and broke this because I wrapped the everything in a function component and didn't use `forwardRef` (because R19 doesn't need to). Oops. 

Fix will be out tomorrow at some point

**Comment by ckifer** [comment]
Released in 2.13.2

Going to close this, lmk if there are issues

**Comment by ckifer** [comment]
Codesandbox which logs the ref for confirmation https://codesandbox.io/p/sandbox/line-chart-width-xaxis-padding-forked-6snqmq
